### ISSUE TITLE
Memory consumption of crf-tagger keeps increasing over time 

### ISSUE THREAD

**Issue by ksboy** [allenai/allennlp#3982]
**Describe the bug**
I am trying to do NER Task using crf-tagger. I pre-tokenized the sentence using bert-tokenizer. The sentence and labels are based on word-piece when are sent to crf-tagger. 
Also, I installed allennlp from master.
Probelm: Memory consumption keeps increasing over time when training.

**To Reproduce**
My training-config is as follows:
`{
  "dataset_reader": {
    "type": "wordpiece_bert_crf_reader",
    "tokenizers":{
      "bert_tokenizer":{
          "type": "whitespace"
      }
  },
    "token_indexers": {
      "transformer": {
        "type": "pretrained_transformer_mismatched",
        "model_name": "/home/mhxia/whou/workspace/bert-base-uncased/",
        "max_length":510
      }
    }
  },
  "train_data_path": "/home/mhxia/whou/workspace/code_repo/space12-ner/data/processed_data/ensemble_data/ner_bert_tokenized/train_1.jsonl",
  "validation_data_path": "/home/mhxia/whou/workspace/code_repo/space12-ner/data/processed_data/ensemble_data/ner_bert_tokenized/dev_1.jsonl",
  "model": {
    "type": "crf_tagger",
    "include_start_end_transitions": false,
    "text_field_embedder": {
      "token_embedders": {
        "transformer": {
          "type": "pretrained_transformer_mismatched",
          "model_name": "/home/mhxia/whou/workspace/bert-base-uncased/",
          "max_length":510
        }
      }
    },
    "encoder": {
      "type": "pass_through",
      "input_dim": 768
    }
  },
  "data_loader": {"batch_size": 2},
  "trainer": {
    "optimizer": "adam",
    "num_epochs": 5,
    "cuda_device":0
  }
}
`

**Expected behavior**
I hope you can fix the bug.

**System (please complete the following information):**
 - OS:  Linux
 - Python version: 3.6.9
 - AllenNLP version:  I installed from master
 - PyTorch version: 1.4.0

**Comment by schmmd** [comment]
@ksboy are you getting an OOME while training or is the memory usage simply increasing?

**Comment by ksboy** [comment]
> OOME
 the memory usage is increasing until OOME
